server/routes/api/stats.py

```diff
@@ -112,26 +112,41 @@ def stats_var_property_wrapper(dcids):
         md = ''
         mprop = ''
         st = ''
+        base = ''
+        mq = ''
         name = dcid
         for triple in triples:
-            if triple['predicate'] == 'measuredProperty':
-                mprop = triple['objectId']
-            if triple['predicate'] == 'populationType':
-                pt = triple['objectId']
-            if triple['predicate'] == 'measurementDenominator':
-                md = triple['objectId']
-            if triple['predicate'] == 'statType':
-                st = triple['objectId']
-            if triple['predicate'] == 'name':
-                name = triple['objectValue']
-            if triple['predicate'] in pvs:
-                pvs[triple['predicate']] = triple['objectId']
+            predicate = triple['predicate']
+            objId = triple.get('objectId', '')
+            objVal = triple.get('objectValue', '')
+            if predicate == 'measuredProperty':
+                mprop = objId
+            if predicate == 'populationType':
+                pt = objId
+            if predicate == 'measurementDenominator':
+                md = objId
+            if predicate == 'statType':
+                st = objId
+            if predicate == 'name':
+                name = objVal
+            if predicate == 'baseDate':
+                base = objVal
+            if predicate == 'measurementQualifier':
+                mq = objId
+            if predicate in pvs:
+                if predicate == 'baseDate':
+                    v = objVal
+                else:
```

Comment:
Here too, should we generalize by looking for objId first, and if not falling back to objVal?

Typically constraint-prop values are IDs, but there is no hard rule that it shouldn't be Text/Date/Number, and this would allow for it.